applications/vessel-history/src/features/vessels/offline-vessels.hook.ts

```diff
@@ -67,8 +69,17 @@ export const useOfflineVesselsAPI = () => {
   )
 
   const dispatchDeleteOfflineVessel = useCallback(
-    (id: string) => {
-      dispatch(deleteOfflineVesselThunk(id))
+    (offlineVessel: OfflineVessel, page: string) => {
+      const now = DateTime.now()
+      const savedOn = DateTime.fromISO(offlineVessel.savedOn);
+      const i = Interval.fromDateTimes(savedOn, now);
+      uaEvent({
+        category: 'Offline Access',
+        action: 'Remove saved vessel for offline view',
+        label: JSON.stringify({ page }),
+        value: Math.floor(i.length('days'))
+      })
```

Comment:
I don't think we should add the GA tracking here since it's directly related to the page or component that we need to track its usage. I'd leave the tracking in the component that calls this dispatch function.